### PR TITLE
fixed error caused by empty strings and warnings deleted

### PR DESCRIPTION
**Pull request by julietpaw**

### CODE REVIEW

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 1]
```diff
@@ -9,143 +9,153 @@
...
+
+		/// <summary>
+		/// Initializes new MailMessage with given parameters. Uses default value 'true' for IsBodyHtml
+		/// </summary>
+		public static MailMessage InitializeMailMessage(Input input)
+		{
+			//split recipients, either by comma or semicolon
+			var separators = new[] { ',', ';' };
+
+			string[] recipients = string.IsNullOrEmpty(input.To)
```
Comment (juliapaw) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs:103]: removed error when string is empty

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 2]
```diff
@@ -9,143 +9,153 @@
...
+		}
+
+		/// <summary>
+		/// Initializes new MailMessage with given parameters. Uses default value 'true' for IsBodyHtml
+		/// </summary>
+		public static MailMessage InitializeMailMessage(Input input)
+		{
+			//split recipients, either by comma or semicolon
+			var separators = new[] { ',', ';' };
+
```
Comment (julietpaw) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs:102]: now the error will not occur if any of these three fields are left blank

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 3]
```diff
@@ -9,143 +9,153 @@
...
-            string folder = Path.GetDirectoryName(filePath);
-            string fileMask = Path.GetFileName(filePath) != "" ? Path.GetFileName(filePath) : "*";
-
-            string[] filePaths = Directory.GetFiles(folder, fileMask);
-
-            return filePaths;
-        }
-    }
-}
+#pragma warning disable IDE1006 // Naming Styles
```
Comment (julietpaw) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs]: pragma IDE1006 used because there was an error related to the Pascal naming convention and only pragma could prevent these warnings

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 4]
```diff
@@ -9,143 +9,153 @@
...
+									output.EmailSent = false;
+									return output;
+								}
+
+								foreach (var fp in allAttachmentFilePaths)
+								{
+									mail.Attachments.Add(new System.Net.Mail.Attachment(fp));
+								}
+							}
+
```
Comment (julietpaw) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs:53]: two if statements connected in one - that was a warning

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 5]
```diff
@@ -9,143 +9,153 @@
...
+		private static ICollection<string> GetAttachmentFiles(string filePath)
+		{
+			string folder = Path.GetDirectoryName(filePath);
+			string fileMask = Path.GetFileName(filePath) != "" ? Path.GetFileName(filePath) : "*";
+
+			string[] filePaths = Directory.GetFiles(folder, fileMask);
+
+			return filePaths;
+		}
+	}
```
Comment (julietpaw) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs:157]: end of pragma

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail.Tests/SendEmailTests.cs [hunk 6]
```diff
@@ -42,7 +42,7 @@ public void EmailTestSetup()
             _filepath = Path.Combine(_localAttachmentFolder, TEST_FILE_NAME);
 
             if (!File.Exists(_filepath))
```
Comment (julietpaw) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail.Tests/SendEmailTests.cs]: there was a warning about empty {}, deleted

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail.Tests/SendEmailTests.cs [hunk 7]
```diff
@@ -42,7 +42,7 @@ public void EmailTestSetup()
...
-                using (File.Create(_filepath)) { }
```
Comment (jefim) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail.Tests/SendEmailTests.cs:45]: Right, so what happens now that the using is without its code block? In the code that had the `{ }` made it clear that we get the FileStream returned by calling File.Create and dispose it. Now without the `{ }` - where does the using block end?..

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/Frends.Smtp.SendEmail.csproj [hunk 8]
```diff
@@ -10,6 +11,22 @@
...
+    <NoWarn>1701;1702;IDE1006</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net6.0|AnyCPU'">
+    <NoWarn>1701;1702;IDE1006</NoWarn>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net6.0|AnyCPU'">
+    <NoWarn>1701;1702;IDE1006</NoWarn>
+  </PropertyGroup>
```
Comment (jefim) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/Frends.Smtp.SendEmail.csproj]: Why do we need all those NoWarns?

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs (hunk 3 above)
Comment (jefim) [reply Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs]: Please enable the warning immediately after the problem place.

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 9]
```diff
@@ -9,143 +9,153 @@
...
+			if (!settings.UseWindowsAuthentication && !string.IsNullOrEmpty(settings.UserName))
+				smtpClient.Credentials = new NetworkCredential(settings.UserName, settings.Password);
+
+			return smtpClient;
+		}
+
+		/// <summary>
+		/// Initializes new MailMessage with given parameters. Uses default value 'true' for IsBodyHtml
+		/// </summary>
+		public static MailMessage InitializeMailMessage(Input input)
```
Comment (jefim) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs]: Why is this changed from private to public?

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 10]
```diff
@@ -9,143 +9,153 @@
...
+			{
+				mail.CC.Add(ccRecipient);
+			}
+			//Add BCC recipients
+			foreach (var bccRecipient in bccRecipients)
+			{
+				mail.Bcc.Add(bccRecipient);
+			}
+
+
```
Comment (jefim) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs]: No need for double newline

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs (hunk 5 above)
Comment (jefim) [reply Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs:157]: Pragma should end much closer to where you disabled it.

**Review by jefim — CHANGES_REQUESTED** [review]

**Comment by jefim** [comment]
Also, please provide a unit test that would prove that the fix is working.

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs (hunk 9 above)
Comment (julietpaw) [reply Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs]: Changed back to private.

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/Frends.Smtp.SendEmail.csproj (hunk 8 above)
Comment (julietpaw) [reply Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/Frends.Smtp.SendEmail.csproj]: Unused ones deleted.

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail.Tests/SendEmailTests.cs (hunk 7 above)
Comment (julietpaw) [reply Frends.SMTP.SendEmail/Frends.Smtp.SendEmail.Tests/SendEmailTests.cs:45]: changed back to {} and added pragma so it won't throw warning

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/Frends.Smtp.SendEmail.csproj (hunk 8 above)
Comment (jefim) [reply Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/Frends.Smtp.SendEmail.csproj]: There should be none at all. You can just delete those PropertyGroups as now they are empty and serve no purpose.

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs [hunk 11]
```diff
@@ -6,146 +6,153 @@
...
+								foreach (var fp in allAttachmentFilePaths)
+								{
+									mail.Attachments.Add(new System.Net.Mail.Attachment(fp));
+								}
+							}
+
+							if (attachment.AttachmentType == AttachmentType.AttachmentFromString
+								&& !string.IsNullOrEmpty(attachment.stringAttachment.FileContent))
+							{
+								mail.Attachments.Add(System.Net.Mail.Attachment.CreateAttachmentFromString
```
Comment (jefim) [inline Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs:57]: Why do you need to specify the whole System.Net.Mail.Attachment.....?

**Review by jefim — CHANGES_REQUESTED** [review]

File: Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs (hunk 11 above)
Comment (julietpaw) [reply Frends.SMTP.SendEmail/Frends.Smtp.SendEmail/SendEmailTask.cs:57]: I need this because there is a class called Attachment and vs confuses the two Attachments, so I guess this is a better way than renaming the class to say mailAttachment?

**Review by jefim — APPROVED** [review]
